libs/applePayments.js

```diff
@@ -1,49 +1,49 @@
-const nconf = require('nconf');
-const iap = require('in-app-purchase');
-const subscriptions = require('../libs/subscriptions');
-const mobilePayments = require('./mobilePayments');
+import iap from 'in-app-purchase';
+import { blocks } from './subscriptions.js';
+import { cancelSubscriptionForUser, scheduleNextCheckForUser } from './mobilePayments.js';
 
 const USERS_BATCH = 10;
-const BASE_URL = nconf.get('BASE_URL');
 
 const INVALID_RECEIPT_ERROR = 21010;
 
-let api = {};
+const api = {};
 
 api.processUser = function processUser (habitrpgUsers, user, jobStartDate, nextScheduledCheck) {
-  let plan = subscriptions.blocks[user.purchased.plan.planId];
+  const plan = blocks[user.purchased.plan.planId];
 
   // Skip users with a blocked account
-  if (user.auth.blocked === true) return;
+  if (user.auth.blocked === true) return false;
 
   if (!plan) {
-    throw new Error(`Plan ${user.purchased.plan.planId} does not exists. User \{user._id}`);
+    throw new Error(`Plan ${user.purchased.plan.planId} does not exists. User ${user._id}`);
   }
   return iap.validate(iap.APPLE, user.purchased.plan.additionalData)
-    .then((response) => {
+    .then(response => {
       if (iap.isValidated(response)) {
-        let purchaseDataList = iap.getPurchaseData(response);
-        for (let index in purchaseDataList) {
-          let subscription = purchaseDataList[index];
-          if (subscription.expirationDate > jobStartDate) {
-            return mobilePayments.scheduleNextCheckForUser(habitrpgUsers, user, subscription, nextScheduledCheck);
+        const purchaseDataList = iap.getPurchaseData(response);
+        for (const index in purchaseDataList) {
+          if (Object.prototype.hasOwnProperty.call(purchaseDataList, index)) {
+            const subscription = purchaseDataList[index];
+            if (subscription.expirationDate > jobStartDate) {
+              return scheduleNextCheckForUser(habitrpgUsers, user, subscription, nextScheduledCheck);
+            }
           }
         }
-        return mobilePayments.cancelSubscriptionForUser(habitrpgUsers, user, "ios");
+        return cancelSubscriptionForUser(habitrpgUsers, user, 'ios');
```

Comment:
It looks like this logic isn't substantially different from what it was before, but I might need you to walk me through it... reading this flow it looks like it always cancels the subscription unless `iap.isValidated` comes up false--that can't be right, can it?

Reply:
or, wait... hitting any `subscription.expirationDate > jobStartDate` returns and breaks the loop... my bad, I thought we had another nested function there!